client/my-sites/posts/post-controls/index.jsx

```diff
@@ -0,0 +1,263 @@
+				className: 'view',
+				href: url.format( parsedUrl ),
+				icon: 'external',
+				onClick: recordPreviewPost,
+				text: translate( 'Preview' ),
+			} );
+
+			if ( canUserPublishPost ) {
+				controls.main.push( {
+					className: 'publish',
+					icon: 'reader',
+					onClick: onPublish,
+					text: translate( 'Publish' ),
+				} );
+			}
+		} else if ( canUserDeletePost ) {
+			controls.main.push( {
+				className: 'restore',
+				icon: 'undo',
+				onClick: onRestore,
+				text: translate( 'Restore' ),
+			} );
+		}
+
+		if ( canUserDeletePost ) {
+			if ( 'trash' === post.status ) {
+				controls.main.push( {
+					className: 'trash is-scary',
+					icon: 'trash',
+					onClick: onDelete,
+					text: translate( 'Delete Permanently' ),
+				} );
+			} else {
+				controls.main.push( {
+					className: 'trash',
+					icon: 'trash',
+					onClick: onTrash,
+					text: translate( 'Trash' ),
+				} );
+			}
+		}
+
+		if ( ( 'publish' === post.status || 'private' === post.status ) && canUserEditPost ) {
+			controls.main.push( {
+				className: 'copy',
+				href: `/post/${ site.slug }?copy=${ post.ID }`,
+				icon: 'clipboard',
+				onClick: recordCopyPost,
+				text: translate( 'Copy' ),
+			} );
+		}
+
+		// More Controls (behind ... more link)
+		if ( controls.main.length > 4 && fullWidth ) {
+			controls.more = controls.main.splice( 4 );
+		} else if ( controls.main.length > 2 && ! fullWidth ) {
+			controls.more = controls.main.splice( 2 );
+		}
+
+		if ( controls.more.length ) {
+			controls.main.push( {
+				className: 'more',
+				icon: 'ellipsis',
+				onClick: this.onShowMore,
+				text: translate( 'More' ),
+			} );
+
+			controls.more.push( {
+				className: 'back',
+				icon: 'chevron-left',
+				onClick: this.onHideMore,
+				text: translate( 'Back' ),
+			} );
+		}
+
+		return controls;
+	}
+
+	render() {
+		const { main, more } = this.getAvailableControls();
+		const classes = classNames( 'post-controls', {
+			'show-more-options': this.state.showMoreOptions,
+			'post-controls--desk-nomore': more <= 2,
+		} );
+
+		return (
+			<div className={ classes }>
+				<ul className="post-controls__pane post-controls__more-options">
+					{ more.map( ( control, index ) =>
+						<PostControl { ...control } key={ index } />
+					) }
+				</ul>
+				<ul className="post-controls__pane post-controls__main-options">
+					{ main.map( ( control, index ) =>
+						<PostControl { ...control } key={ index } />
+					) }
+				</ul>
+			</div>
+		);
+	}
+}
+
+const mapStateToProps = ( state, { site, post } ) => {
+	const siteId = site && site.ID ? site.ID : null;
+	return {
+		canUserDeletePost: canCurrentUser( state, siteId, 'delete_posts' ),
+		canUserEditPost: canCurrentUser( state, siteId, 'edit_posts' ),
+		canUserPublishPost: canCurrentUser( state, siteId, 'publish_posts' ),
+		isPublicizeEnabled: isPublicizeEnabled( state, siteId, post.type ),
+	};
+};
+
+const mapDispatchToProps = {
+	recordCopyPost: () => composeAnalytics(
```

Comment:
Ah yup, that's the analytics error I've fixed in the other PR and i didn't have the chance to rebase yet.